### PR TITLE
Add support for explicit table-level locks

### PR DESCRIPTION
**Pull request by Photonios**

### CODE REVIEW

File: docs/source/locking.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
+.. include:: ./snippets/postgres_doc_links.rst
+
+.. _locking_page:
+
+Locking
+=======
+
+`Explicit table-level locks`_ are supported through the :meth:`psqlextra.locking.postgres_lock_model` and :meth:`psqlextra.locking.postgres_lock_table` methods. All table-level lock methods are supported.
+
+Locks are always bound to the current transaction and are released when the transaction is comitted or rolled back. There is no support (in PostgreSQL) for explicitly releasing a lock.
```
Comment (sewi-cpan) [inline docs/source/locking.rst:10]:
```suggestion
Locks are always bound to the current transaction and are released when the transaction is committed or rolled back. There is no support (in PostgreSQL) for explicitly releasing a lock.
```

File: docs/source/locking.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+.. code-block:: python
+
+    from django.db import transaction
+
+    from psqlextra.locking import PostgresTableLockMode, postgres_lock_table
+
+    with transaction.atomic(durable=True):
+        postgres_lock_model(MyModel, PostgresTableLockMode.EXCLUSIVE)
+
+    # locks are released here, when the transaction comitted
```
Comment (sewi-cpan) [inline docs/source/locking.rst:34]:
```suggestion
    # locks are released here when the transaction committed
```

File: psqlextra/locking.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            Type of lock to acquire.
+
+        schema_name:
+            Optionally, the unquoted name of the schema
+            the table to lock is in. If not specified,
+            the table name is resolved by PostgreSQL
+            using it's ``search_path``.
+
+        using:
+            Name of the database alias to use.
```
Comment (sewi-cpan) [inline psqlextra/locking.py:48]:
```suggestion
            Optional name of the database alias to use.
```

File: psqlextra/locking.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            the table to lock is in. If not specified,
+            the table name is resolved by PostgreSQL
+            using it's ``search_path``.
+
+            Django models always reside in the default
+            ("public") schema. You should not specify
+            this unless you're doing something special.
+
+        using:
+            Name of the database alias to use.
```
Comment (sewi-cpan) [inline psqlextra/locking.py:92]:
```suggestion
            Optional name of the database alias to use.
```

File: tests/test_locking.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    with connection.cursor() as cursor:
+        return connection.introspection.get_table_locks(cursor)
+
+
+@pytest.mark.django_db(transaction=True)
+def test_postgres_lock_table(mocked_model):
+    lock_signature = (
+        "public",
+        mocked_model._meta.db_table,
+        "AccessExclusiveLock",
```
Comment (sewi-cpan) [inline tests/test_locking.py:35]: What using `@pytest.mark.parametrize` with `PostgresTableLockMode` to ensure that all lock modes are working?

File: tests/test_locking.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        postgres_lock_table(
+            table_name, PostgresTableLockMode.EXCLUSIVE, schema_name=schema_name
+        )
+        assert lock_signature in get_table_locks()
+
+    assert lock_signature not in get_table_locks()
+
+
+@pytest.mark.django_db(transaction=True)
+def test_postgres_lock_mode(mocked_model):
```
Comment (sewi-cpan) [inline tests/test_locking.py:70]:
```suggestion
def test_postgres_lock_model(mocked_model):
```

File: psqlextra/locking.py (hunk 3 above)
Comment (Photonios) [reply psqlextra/locking.py]: I've addressed this by making the description for this parameter across the code base consistent:

> Optional name of the database connection to use.

**Review by sewi-cpan — APPROVED** [review]

**Pull request merged**